simphony/testing/abc_check_engine.py

```diff
@@ -177,15 +177,17 @@ def test_delete_dataset(self):
             name = "test_" + str(i)
             engine.add_dataset(self.create_dataset(name=name))
 
+        datasets_and_names = [(ds, ds.name) for ds in engine.iter_datasets()]
+
         # delete each of the datasets
-        for ds in engine.iter_datasets():
-            engine.remove_dataset(ds.name)
+        for (dataset, name) in datasets_and_names:
+            engine.remove_dataset(name)
             # test that we can't get deleted datasets
             with self.assertRaises(ValueError):
-                engine.get_dataset(ds.name)
+                engine.get_dataset(name)
             # test that we can't use the deleted datasets
             with self.assertRaises(Exception):
-                self.compare_dataset(ds, ds)
+                self.compare_dataset(dataset, dataset)
```

Comment:
I think the test is okay as it is.

Lines 189-190 is to check that we get an error when we try to use the deleted `dataset`.   From line 180, we know that  `dataset` is a *proxy of the dataset named name that is stored internally in the Engine.*  In line 184, we know that the dataset with this name was removed from the engine.  Therefore, the proxy 'dataset' should no longer work.  It should throw an exception if it is used (i.e. data.name, data.iter_nodes ... or other things that will be called in `self.compare_dataset`).